src/components/realm-selector/RealmSelector.tsx

```diff
@@ -1,62 +1,120 @@
 import React, { useState, useContext, useEffect } from "react";
 import { useHistory } from "react-router-dom";
 import { Realm } from "../../realm/models/Realm";
+import { RealmContext } from "../realm-context/RealmContext";
 
 import {
   Dropdown,
   DropdownToggle,
   DropdownItem,
   Button,
   Divider,
+  SplitItem,
+  Split,
+  ContextSelector,
+  ContextSelectorItem,
 } from "@patternfly/react-core";
 
 import style from "./realm-selector.module.css";
+import { CheckIcon } from "@patternfly/react-icons";
 
 type RealmSelectorProps = {
-  realm: string;
   realmList: Realm[];
 };
 
-export const RealmSelector = ({ realm, realmList }: RealmSelectorProps) => {
+export const RealmSelector = ({ realmList }: RealmSelectorProps) => {
+  const { realm, setRealm } = useContext(RealmContext);
   const [open, setOpen] = useState(false);
+  const [search, setSearch] = useState("");
+  const [filteredItems, setFilteredItems] = useState(realmList);
   const history = useHistory();
-  const [currentRealm, setCurrentRealm] = useState(realm);
+
+  const toUpperCase = (realmName: string) =>
+    realmName.charAt(0).toUpperCase() + realmName.slice(1);
+
+  const RealmText = ({ value }: { value: string }) => (
+    <Split>
+      <SplitItem isFilled>{toUpperCase(value)}</SplitItem>
+      <SplitItem>{value === realm && <CheckIcon />}</SplitItem>
+    </Split>
+  );
```

Comment:
Wouldn't be simpler and cleaner to avoid deconstruction in TypeScript land?

The datatype already defines the structure of the data, and deconstruction habits create larger code base and harder to use refactoring tools from IDEs since they require runtime execution for rename and so on, but static information like datatypes don't.

🤷‍♂️

```suggestion
  const RealmText = (props: { value: string }) => (
    <Split>
      <SplitItem isFilled>{toUpperCase(props.value)}</SplitItem>
      <SplitItem>{props.value === realm && <CheckIcon />}</SplitItem>
    </Split>
  );
```

Reply:
I don't think that is true, when deconstructing the properties they will still be checked in the example above if I would to type `({ something }: TheType)` the check will be if something is defined in `TheType`

Reply:
Let's agree to disagree.

Personally, such practices end up doubling the codebase size in TypeScript land (deconstruction + type spec everywhere), and make it harder to refactor in the future, but I guess we have different experiences.

🤷‍♂️ 

## Deconstructing

![Sep-15-2020 02-14-12](https://user-images.githubusercontent.com/4237280/93172645-1d208880-f6e0-11ea-92a9-59da37eaeb7f.gif)

## Non-deconstructing

![Sep-15-2020 02-14-29](https://user-images.githubusercontent.com/4237280/93172649-1eea4c00-f6e0-11ea-9c24-77b127b718cd.gif)

Reply:
Are you using the argument that because it's a newer feature we should not use it? By that logic we should stick to angular.

Reply:
In the your screenshots you are refactoring the name of the property in the first one and the other the name of the variable, I don't see why we would not use deconstructing just because of that.

Reply:
@yordis If you have a problem with destructuring in TypeScript then the place to discuss that is over at the TypeScript project.

Reply:
Has nothing to do with TypeScript itself (since deconstruction for the sake of, still an over abused feature that meant to be use for something else) I think you misinterpreted the feedback.

Regardless, no point in having a discussion about it, it is your project and your guidelines, deconstructing will be 🤷